### PR TITLE
refactor(server): clean up lib/server.js

### PR DESCRIPTION
**Pull request by hiroppy**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [x] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

no

### Motivation / Use-Case

Reduce lines in constructor, delete eslint comments, fix typo, and etc..

### Breaking Changes

no

### CODE REVIEW

**Comment by evilebottnawi** [comment]
You can send a PR with fixing lint problems and typos, it is not breaking change

**Comment by hiroppy** [comment]
I gave up changing property names. I'll change these when we release as semver-major:)

File: lib/Server.js [hunk 1]
```diff
@@ -641,11 +643,10 @@ class Server {
   }
 
   setContentHeaders(req, res, next) {
-    if (this.headers) {
-      // eslint-disable-next-line
-      for (const name in this.headers) {
-        res.setHeader(name, this.headers[name]);
-      }
+    if (this.options.headers) {
```
Comment (evilebottnawi) [inline lib/Server.js]: Let's wait this too because developers can change `this.headers` in runtime and it is break logic

File: lib/Server.js [hunk 2]
```diff
@@ -625,6 +513,120 @@ class Server {
...
+        // eslint-disable-next-line global-require
+        this.listeningApp = require('spdy').createServer(
+          this.options.https,
+          this.app
+        );
+      }
+    } else {
+      this.listeningApp = http.createServer(this.app);
+    }
+  }
```
Comment (evilebottnawi) [inline lib/Server.js]: I think we should rename this method on `createServer`, also we can refactor this on two methods:
- setupHttps (logic for https)
- createServer (logic for `http.createServer`)

**Comment by hiroppy** [comment]
@evilebottnawi fixed https://github.com/webpack/webpack-dev-server/compare/781525c4150e807a6611d5833b3a3b56e93e3fa4..7852e0762f074d6186a256bb9ddc82cab5560537

File: lib/Server.js [hunk 3]
```diff
@@ -625,6 +513,126 @@ class Server {
...
+        );
+      } else {
+        // if the normal https server gets this option, it will not affect it.
+        this.options.https.spdy = {
+          protocols: ['h2', 'http/1.1'],
+        };
+      }
+    }
+
+    this.createServer();
```
Comment (evilebottnawi) [inline lib/Server.js]: Let's move this after ` this.setupHttps();` (no in method)

Comment (hiroppy) [reply lib/Server.js]: fixed

**Comment by evilebottnawi** [comment]
Let's wait CI green and feel free to merge

**Pull request merged**